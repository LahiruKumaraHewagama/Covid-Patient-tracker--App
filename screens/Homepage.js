import React from 'react';
import { StyleSheet, Text, View, SafeAreaView, Image, TouchableOpacity } from 'react-native';
import * as Animatable from 'react-native-animatable';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

function Homepage ({ navigation }) {
  return (
    <SafeAreaView style={styles.container}>
      <View style={StyleSheet.header}>
        <Image style={styles.logo} source={require('../assets/logo.png')}/>
      </View>
      <Animatable.View style={styles.footer} animation="fadeInUpBig">
          <Text style={styles.title}>Stay Safe!</Text>
          <AppButton onPress={() => navigation.navigate('Signin')} title={'Get Started'}/>
      </Animatable.View>
    </SafeAreaView>
  );
}

const AppButton = ({ onPress, title }) => (
  <TouchableOpacity onPress={onPress} style={styles.button}>
    <Text style={styles.buttonText}>{title}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  },
  header: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center'
  },
  logo: {
    width: 350,
    height: 150,
    marginTop: 150,
    marginBottom: 120
  },
  footer: {
    flex: 1,
    backgroundColor: '#009387',
    borderTopLeftRadius: 30,
    borderTopRightRadius: 30,
    paddingVertical: 70,
    paddingHorizontal: 40
  },
  title: {
    color: '#fff',
    fontSize: 35,
    fontWeight: 'bold',
    paddingBottom: 65
  },
  button: {
    paddingVertical: 10,
    paddingHorizontal: 50,
    backgroundColor: '#fff',
    borderRadius: 50,
    alignItems: 'center',
    alignSelf: 'flex-end'
  },
  buttonText: {
    color: '#000',
    fontWeight: 'bold',
    fontSize: 20
  }
});

export default Homepage;