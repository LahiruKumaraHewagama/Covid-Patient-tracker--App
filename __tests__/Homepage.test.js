import React from "react";
import Homepage from "../screens/Homepage";
import { render, fireEvent } from '@testing-library/react-native';
import { act, create } from "react-test-renderer";

jest.useFakeTimers()
const navigation = { navigate: jest.fn() }
const tree = create(<Homepage navigation={navigation} />);

test('snapshot home page ', () => {
    expect(tree).toMatchSnapshot();
});

test('navigate to signin screen', () => {
    const getStartBtn = tree.root.findByProps({ testID: 'getStart' }).props;
    getStartBtn.onPress();
    expect(navigation.navigate).toBeCalledWith('Signin');
})






// it('Should Error Message ' , ()=>{

// const {getByText }=render(<Homepage/>);
// const  startBtn = getByText('Get Started');

// fireEvent.press(startBtn);

// const  errorMessage =getByText("Email can't be empty !");
// expect(errorMessage).not.toBeNull();
// }

// )

// import React from "react";
// import Homepage from "../screens/Homepage";
// import renderer from "react-test-renderer";
// import {Button, Text, TextInput, View} from 'react-native'
// import {fireEvent, render, waitFor} from '@testing-library/react-native'

// jest.useFakeTimers()

// test('Home page renders correctly', () =>{
//     const tree = renderer.create(<Homepage/>).toJSON();
//     expect(tree).toMatchSnapshot();
// });


// it ("Invalid input messages" , () =>{
//     const {getAllByText} = render(<Homepage/>);
//     fireEvent.press(getAllByText("Get Started"))

// })