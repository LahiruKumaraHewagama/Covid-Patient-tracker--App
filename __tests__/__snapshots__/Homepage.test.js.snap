// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot home page  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 150,
          "marginBottom": 120,
          "marginTop": 150,
          "width": 300,
        }
      }
    />
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#009387",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 1,
        "opacity": 0,
        "paddingHorizontal": 40,
        "paddingVertical": 70,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 35,
          "fontWeight": "bold",
          "paddingBottom": 65,
        }
      }
    >
      Stay Safe  SriLanka !
    </Text>
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#fff",
            "borderColor": "#20d1ce",
            "borderRadius": 10,
            "borderWidth": 3,
            "elevation": 12,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 5,
            "shadowColor": "#fff",
            "shadowOpacity": 80,
            "shadowRadius": 2,
          }
        }
        testID="getStart"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#009387",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Get Started
          </Text>
          <View />
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
